proto/p4/p4runtime.proto

```diff
@@ -136,6 +136,14 @@ message TableEntry {
   // this is stored, as long as it is returned with the rest of the entry in
   // a Read RPC.
   uint64 controller_metadata = 5;
+  // meter_config and counter_data are convenience fields that enable the
+  // controller to configure the direct resources associated with the table at
+  // the same time as a match-action entry is inserted or modified. When
```

Comment:
Suggest adding:
If the table does not have a direct resource, then setting the direct resource field in any table write operation will return an error. If the table has a direct resource and a table entry is inserted without setting the direct resource field(s), the field(s) are assigned their default values.